src/main/java/uk/gov/pay/products/service/ProductFinder.java

```diff
@@ -92,26 +95,31 @@ public Boolean deleteByGatewayAccountIdAndExternalId(Integer gatewayAccountId, S
         return productDao.findByGatewayAccountId(gatewayAccountId)
                 .stream()
                 .map(ProductEntity::toProduct)
-                .collect(Collectors.toList())
+                .map(linksDecorator::decorate)
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    @Transactional
+    public List<Product> findByGatewayAccountIdAndType(Integer gatewayAccountId, ProductType type) {
```

Comment:
Personally I'd prefer most methods in ProductFinder to be simply `find` with overloaded methods. But it's not a biggie.

Reply:
I was just following the existing conventions.